rhino/src/main/java/org/mozilla/javascript/IRFactory.java

```diff
@@ -1474,7 +1503,7 @@ private Node createForIn(
             if (destructuring != -1) {
                 assign =
                         parser.createDestructuringAssignment(
-                                declType, lvalue, id, (AstNode node) -> transform(node));
```

Comment:
Given we have a number of places where you've had to add `null` to the argument list here would it be worth reducing the noise by defining 
```java
Node createDestructuringAssignment(int, type, Node left, Node right, Transformer transformer) {
    return createDestructuringAssignment(type, left, right, null, transformer);
}
```